internal/ldconfig/ldconfig.go

```diff
@@ -179,22 +187,30 @@ func (l *Ldconfig) filterDirectories(configFilePath string, directories ...strin
 	return filtered, nil
 }
 
-// createLdsoconfdFile creates a file at /etc/ld.so.conf.d/.
-// The file is created at /etc/ld.so.conf.d/{{ .pattern }} using `CreateTemp` and
-// contains the specified directories on each line.
-func createLdsoconfdFile(pattern string, dirs ...string) error {
+// createLdsoconfdFile creates a ldso conf file containing the specified directories on each line.
+//
+// The file is either created at `topLevelLdsoconfFilePath` if no file exists there, or as a
+// drop-in in `ldsoconfdDir` using `CreateTemp` and `pattern`. The function assumes that an
+// existing top-level ldso conf file includes `ldsoconfdDir`.
+func createLdsoconfdFile(topLevelLdsoconfFilePath, ldsoconfdDir, pattern string, dirs ...string) error {
 	if len(dirs) == 0 {
 		return nil
 	}
 
-	ldsoconfdDir := "/etc/ld.so.conf.d"
-	if err := os.MkdirAll(ldsoconfdDir, 0755); err != nil {
-		return fmt.Errorf("failed to create ld.so.conf.d: %w", err)
-	}
-
-	configFile, err := os.CreateTemp(ldsoconfdDir, pattern)
-	if err != nil {
-		return fmt.Errorf("failed to create config file: %w", err)
+	var configFile *os.File
+	if _, err := os.Stat(topLevelLdsoconfFilePath); os.IsNotExist(err) {
+		configFile, err = os.OpenFile(topLevelLdsoconfFilePath, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0644)
```

Comment:
Since we also rely on drop-in files to enable CUDA forward compatibility, would it be better to create an `ld.so.conf` file with the following contents if it doesn't exist:
```
$ cat /etc/ld.so.conf
include /etc/ld.so.conf.d/*.conf
```
Then we don't have to also have a special case for the compat libraries.